### PR TITLE
restructure foundry contract directory

### PR DESCRIPTION
**Pull request by KelvinThai**
# Description

This PR restructure the contract folder, fix github actions

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [ ] Should publish Docker image

### CODE REVIEW

**Review by nick-bisonai — CHANGES_REQUESTED** [review]
- I assume the initial request was to put projects inside `contracts` folder rather than having scattered path as `contracts-v0.1` and `contracts-v0.2`.

so structure would be as followed if I understood correctly
```
contracts
├─ v0.1 (hardhat project)
├─ v0.2 (foundry project)
```
- I think it'd unnecessary to keep `v0.1` path inside `scripts`, `src`, and `test` folder since the versioning is done in parent path
- It'd be better to keep all github workflow files inside `.gitgub/workflows` folder rather than having it separated in project folder

**Comment by martinkersner** [comment]
> I think we alr agreed up on this structure in the sync meeting. Anyway, it does not make much impact image

Nick's approach is also fine, but let's keep it as it is in this PR. Later, we can do update it again :)

File: vrf/README.md [hunk 1]
```diff
@@ -44,3 +44,5 @@ Some parts of code were inspired by or copied from
 
 - [node-ecrg](https://github.com/KenshiTech/node-ecvrf)
 - [vrf-ts-256](https://github.com/cbrpunks/vrf-ts-256)
+
+###
```
Comment (martinkersner) [inline vrf/README.md:48]: @KelvinThai Can you please remove this?

```suggestion

```

**Review by martinkersner — APPROVED** [review]
LGTM!

File: contracts-v0.2/.github/workflows/test.yml
Comment (nick-bisonai) [inline contracts-v0.2/.github/workflows/test.yml:1]: is there a specific reason for this file to stay in this project path?
if not, please move the github action file inside `contracts-v0.2` folder

Comment (KelvinThai) [reply contracts-v0.2/.github/workflows/test.yml:1]: Agreed, let's keep all workflow files in the root project directory

**Review by nick-bisonai — APPROVED** [review]

**Pull request merged**